packages/orchestration/src/orchestration-response.ts

```diff
@@ -95,16 +96,11 @@ export class OrchestrationResponse {
     return this.findChoiceByIndex(choiceIndex)?.message;
   }
 
-  private getChoices() {
+  private getChoices(): LlmChoice[] {
     return this.data.orchestration_result.choices;
   }
 
-  private findChoiceByIndex(index: number) {
-    // TODO: replace cast with LLMChoice[] after the bug in orchestration, where
-    // 'role' in ResponseChatMessage is optional when it should be mandatory, is fixed.
-    // https://github.com/SAP/ai-sdk-js-backlog/issues/306
-    return (this.getChoices() as any).find(
-      (c: { index: number }) => c.index === index
-    );
+  private findChoiceByIndex(index: number): LlmChoice | undefined {
```

Comment:
[req] As we export this function in orchestration streaming, and foundation models, here we should also expose it publicly. The reason why we hide this before was because it was not really usable.